app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -1,29 +1,32 @@
 class OmniauthCallbacksController < Devise::OmniauthCallbacksController
   # Don't need a policy for this since this is our sign up/in route
   include Devise::Controllers::Rememberable
-  def self.provides_callback_for(provider)
-    # raise ApplicationConfig["omniauth.auth"].to_yaml
-    class_eval %{
-      def #{provider}
-        cta_variant = request.env["omniauth.params"]['state'].to_s
-        @user = AuthorizationService.new(request.env["omniauth.auth"], current_user, cta_variant).get_user
-        if @user.persisted? && @user.valid?
-          remember_me(@user)
-          sign_in_and_redirect @user, event: :authentication
-          set_flash_message(:notice, :success, kind: "#{provider}".capitalize) if is_navigational_format?
-        elsif @user.persisted? && @user.errors.full_messages.join(", ").include?("username has already been taken")
-          redirect_to "/settings?state=previous-registration"
-        else
-          session["devise.#{provider}_data"] = request.env["omniauth.auth"]
-          user_errors = @user.errors.full_messages
-          flash[:alert] = user_errors
-          redirect_to new_user_registration_url
-        end
-      end
-    }
+
+  def twitter
+    callback_for('twitter')
+  end
+
+  def github
+    callback_for('github')
   end
 
-  %i[twitter github].each do |provider|
-    provides_callback_for provider
+  private
+
+  def callback_for(provider)
+    raise '9999999999999999999999'
```

Comment:
@abraham Yes this is intentional, tests don't work locally for me.
I wanted to know which tests were covering this line, so I created this commit and let the CI run the specs.
For my surprise nothing break, I'll create a test for covering these lines before refactoring.
The PR is not ready for review yet.

Reply:
Might I suggest a descriptive error in the future like `raise 'Checking if run on CI'` or similar.

This might be difficult to test with its reliance on external services.